File: src/main/java/aurocosh/divinefavor/common/network/message/client/syncing/MessageSyncWindLeash.java
package aurocosh.divinefavor.common.network.message.client.syncing;

import aurocosh.divinefavor.DivineFavor;
import aurocosh.divinefavor.common.custom_data.living.LivingData;
import aurocosh.divinefavor.common.custom_data.living.data.wind_leash.WindLeashData;
import aurocosh.divinefavor.common.network.base.WrappedClientMessage;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.util.math.Vec3d;
import net.minecraftforge.fml.relauncher.Side;
import net.minecraftforge.fml.relauncher.SideOnly;

public class MessageSyncWindLeash extends WrappedClientMessage {
    public double x;
    public double z;

    public MessageSyncWindLeash() {
    }

    public MessageSyncWindLeash(double x, double z) {
        this.x = x;
        this.z = z;
    }

    @Override
    @SideOnly(Side.CLIENT)
    protected void handleSafe() {
        EntityPlayer player = DivineFavor.proxy.getClientPlayer();
        WindLeashData windLeash = LivingData.INSTANCE.get(player).getWindLeashData();
        windLeash.setVector(new Vec3d(x, 0, z));
    }
}

File: src/main/kotlin/aurocosh/divinefavor/common/network/message/client/syncing/MessageSyncWindLeash.kt
package aurocosh.divinefavor.common.network.message.client.syncing

import aurocosh.divinefavor.DivineFavor
import aurocosh.divinefavor.common.lib.extensions.divineLivingData
import aurocosh.divinefavor.common.network.message.base.DivineClientMessage
import net.minecraft.util.math.Vec3d
import net.minecraftforge.fml.relauncher.Side
import net.minecraftforge.fml.relauncher.SideOnly

class MessageSyncWindLeash : DivineClientMessage {
    var x: Double = 0.0
    var z: Double = 0.0

    constructor()

    constructor(vec3d: Vec3d) {
        this.x = vec3d.x
        this.z = vec3d.z
    }

    @SideOnly(Side.CLIENT)
    override fun handleSafe() {
        if (!DivineFavor.proxy.hasClientPlayer)
            return
        val player = DivineFavor.proxy.clientPlayer
        val windLeash = player.divineLivingData.windLeashData
        windLeash.vector = Vec3d(x, 0.0, z)
    }
}
